pynicotine/shares.py

```diff
@@ -744,6 +725,69 @@ def get_compressed_shares_message(self, share_type):
 
         return self.compressed_shares.get(share_type)
 
+    def get_shared_folders(self):
+
+        shared_folders = config.sections["transfers"]["shared"]
+        shared_folders_buddy = config.sections["transfers"]["buddyshared"]
+
+        return shared_folders, shared_folders_buddy
+
+    def get_normalized_virtual_name(self, virtual_name, shared_folders=None):
+
+        if shared_folders is None:
+            public_shares, buddy_shares = self.get_shared_folders()
+            shared_folders = (public_shares + buddy_shares)
+
+        # Provide a default name for root folders
+        if not virtual_name:
+            virtual_name = "Shared"
+
+        # Remove slashes from share name to avoid path conflicts
+        virtual_name = virtual_name.replace("/", "_").replace("\\", "_").strip(' "')
+        new_virtual_name = str(virtual_name)
+
+        # Check if virtual share name is already in use
+        counter = 1
+        while new_virtual_name in (x[0] for x in shared_folders):
+            new_virtual_name = f"{virtual_name}{counter}"
+            counter += 1
+
+        return new_virtual_name
+
+    def add_share(self, folder_path, group_name="public", virtual_name=None, share_groups=None, validate_path=True):
+
+        if share_groups is None:
+            share_groups = self.get_shared_folders()
+
+        public_shares, buddy_shares = share_groups
+
+        if folder_path in (x[1] for x in public_shares + buddy_shares):
+            return None
+
+        if validate_path and not os.access(encode_path(folder_path), os.R_OK):
+            return None
+
+        virtual_name = core.shares.get_normalized_virtual_name(
+            virtual_name or os.path.basename(folder_path), shared_folders=(public_shares + buddy_shares)
+        )
+        shares = buddy_shares if group_name == "buddy" else public_shares
+
+        shares.append((virtual_name, folder_path))
+        return virtual_name
+
+    def remove_share(self, virtual_name_or_folder_path, share_groups=None):
+
+        if share_groups is None:
+            share_groups = self.get_shared_folders()
+
+        for shares in share_groups:
+            for virtual_name, folder_path in shares:
+                if virtual_name_or_folder_path in (virtual_name, folder_path):
+                    shares.remove((virtual_name, folder_path))
+                    return True
+
+        return False
```

Comment:
Consider returning the `folder_path` or `None` to print a more helpful CLI output.

Reply:
> After removal using the `/unshare` command, it doesn't seem possible to `/share` it again.

This happened due to group_name and folder_path arguments being incorrectly swapped in the share command.